local_groups/views.py

```diff
@@ -49,18 +55,32 @@ class EventCreateView(LoginRequiredMixin, SuccessMessageMixin, CreateView):
     ''' if settings.EVENT_AUTO_APPROVAL else '''
     Your event was created successfully and is now being reviewed by our team.
     '''
+    permission_required = 'local_groups.add_event'
+
+    def get_local_group(self):
+        local_group = None
+        user = self.request.user
 
-    # Check if user is a group leader and has a valid bsd cons_id
+        if hasattr(user, 'localgroupprofile'):
```

Comment:
I liked your suggestion of putting this into a reusable method. Although I did not implement either because was out of time.

Reply:
Same, we can if needed in future, I didn't want to do because wasn't sure what you were doing.